src/common/vector/auxiliary_buffer.cpp

```diff
@@ -34,18 +34,25 @@ list_entry_t ListAuxiliaryBuffer::addList(uint64_t listSize) {
 }
 
 void ListAuxiliaryBuffer::resizeDataVector(ValueVector* dataVector) {
+    auto buffer = std::make_unique<uint8_t[]>(capacity * dataVector->getNumBytesPerValue());
+    memcpy(buffer.get(), dataVector->valueBuffer.get(), size * dataVector->getNumBytesPerValue());
+    dataVector->valueBuffer = std::move(buffer);
+    dataVector->nullMask->resize(capacity);
     // If the dataVector is a struct vector, we need to resize its field vectors.
     if (dataVector->dataType.getPhysicalType() == PhysicalTypeID::STRUCT) {
-        auto fieldVectors = StructVector::getFieldVectors(dataVector);
-        for (auto& fieldVector : fieldVectors) {
-            resizeDataVector(fieldVector.get());
-        }
-    } else {
-        auto buffer = std::make_unique<uint8_t[]>(capacity * dataVector->getNumBytesPerValue());
-        memcpy(
-            buffer.get(), dataVector->valueBuffer.get(), size * dataVector->getNumBytesPerValue());
-        dataVector->valueBuffer = std::move(buffer);
-        dataVector->nullMask->resize(capacity);
+        resizeStructDataVector(dataVector);
```

Comment:
should we implement a resize() function for each vector like how we do copyToRowData(). Because the same problem same to also exists for LIST of LIST